src/lipalightninglib.udl

```diff
@@ -538,9 +538,12 @@ interface MnemonicError {
 
 [Error]
 interface DecodeInvoiceError {
+    PermanentFailure(string msg);
     ParseError(string msg);
     SemanticError(string msg);
     NetworkMismatch(Network expected, Network found);
+    PayingToSelf();
+    InvoiceExpired();
```

Comment:
I'm not sure we want to fail decoding if the invoice is expired. If I try to pay an expired invoice, I could still be interested in checking out the amount that the payee requested. Indeed this a detail, so if you think it isn't worth it to account for this potential use case, feel free to ignore this.